pyam/utils.py

```diff
@@ -108,7 +109,9 @@ def write_sheet(writer, name, df, index=False):
 
 def read_pandas(path, default_sheet='data', *args, **kwargs):
     """Read a file and return a pandas.DataFrame"""
-    if path.endswith('csv'):
+    if isinstance(path, Path) and path.suffix == '.csv':
+        df = pd.read_csv(path, *args, **kwargs)
+    elif isstr(path) and path.endswith('csv'):
```

Comment:
given the casting above, is it even possible to get here?

Reply:
That's what I thought, then this one bit me... https://github.com/IAMconsortium/pyam/blob/33156c62a95465785d7fc08b80fa3408ebe3eb36/pyam/core.py#L1640

And it got me thinking that some users may be calling that function directly for convenience in other situations, so rather stay backward-compatible...

Reply:
also good

Reply:
On second thought, I think you're right... Avoid duplication within the function and just do the casting in the other location where the function is called...